src/components/Datepicker/sgds-datepicker.ts

```diff
@@ -0,0 +1,307 @@
+      // Get the formattedDate value for the selected dates
+      const formattedDate = this.makeInputValueString(
+        this.selectedDateRange[0],
+        this.selectedDateRange[1],
+        this.dateFormat
+      );
+
+      this.value = formattedDate;
+
+      this.emit("sgds-change-date");
+    } else {
+      // If selectedDateRange is empty, emit sgds-change-date event with an empty string
+      this.value = "";
+      this.emit("sgds-change-date");
+    }
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    // Remove the click event listener from the document
+    document.removeEventListener("click", (event: MouseEvent) => this._handleClickOutOfElement(event, this));
+  }
+
+  /** @internal */
+  private makeInputValueString = (startDate: Date, endDate: Date, dateFormat: string) => {
+    if (!startDate && !endDate) return "";
+
+    const formatDate = (date: Date) => {
+      if (!(date instanceof Date) || isNaN(date.getTime())) {
+        return ""; // Return an empty string if the date is not a valid Date object
+      }
+
+      const month = date.getMonth() + 1;
+      const day = date.getDate();
+      const separator = "/";
+      const formattedMonth = month > 9 ? month.toString() : `0${month}`;
+      const formattedDay = day > 9 ? day.toString() : `0${day}`;
+
+      return dateFormat === "YYYY/MM/DD"
+        ? `${date.getFullYear()}${separator}${formattedMonth}${separator}${formattedDay}`
+        : dateFormat === "DD/MM/YYYY"
+        ? `${formattedDay}${separator}${formattedMonth}${separator}${date.getFullYear()}`
+        : `${formattedMonth}${separator}${formattedDay}${separator}${date.getFullYear()}`;
+    };
+
+    if (startDate && endDate) {
+      return `${formatDate(startDate)} - ${formatDate(endDate)}`;
+    }
+
+    if (startDate) {
+      return formatDate(startDate);
+    }
+
+    return "";
+  };
+
+  /** @internal */
+  private handleSelectDates(event: CustomEvent<Date[]>) {
+    const newSelectedDates = event.detail;
+    if (this.mode === "range") {
+      // Sort the newSelectedDates array in ascending order
+      newSelectedDates.sort((a: Date, b: Date) => a.getTime() - b.getTime());
+
+      this.selectedDateRange = newSelectedDates;
+      if (this.selectedDateRange.length === 2) {
+        this.hideMenu();
+      }
+      this.displayDateInput = newSelectedDates.length > 0 ? newSelectedDates[0] : undefined;
+    } else if (this.mode === "single") {
+      this.selectedDateRange = [newSelectedDates[0]];
+
+      if (this.selectedDateRange.length === 1) {
+        this.hideMenu();
+      }
+    }
+
+    // Get the formattedDate value for the selected dates
+    const formattedDate = this.makeInputValueString(
+      this.selectedDateRange[0],
+      this.selectedDateRange[1],
+      this.dateFormat
+    );
+
+    // Set formattedDate value as the new value for sgds-input
+    this.value = formattedDate;
+
+    this.emit("sgds-change-date");
```

Comment:
since you are emitting sgds-change-date at several places and it all emits because value has change 

you can refactor to use the watch decorator, to watch for changes to value and then emit the event inside that method. That way you maintain one method for the same event